cmd/provisioning/lookup.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provisioning
+
+import (
+	"fmt"
+
+	"github.com/apex/log"
+	"github.com/spf13/cobra"
+
+	"github.com/cisco-open/fsoc/output"
+	"github.com/cisco-open/fsoc/platform/api"
+)
+
+func newCmdLookup() *cobra.Command {
+
+	var lookupCmd = &cobra.Command{
+		Use:              "lookup-tenant",
+		Short:            "Lookup for a tenant by vanity URL",
+		Long:             `Check whether tenant exist and return tenant Id if it does.`,
+		Example:          `  provisioning lookup-tenant --vanityUrl=fsoc-test.saas.appd-test.com`,
+		Args:             cobra.ExactArgs(0),
+		Run:              lookup,
+		TraverseChildren: true,
+	}
+
+	vanityUrlFlag := "vanityUrl"
+	lookupCmd.Flags().String(vanityUrlFlag, "", "Vanity URL without a scheme. Provide only domain part.")
+	_ = lookupCmd.MarkFlagRequired(vanityUrlFlag)
+
+	return lookupCmd
+}
+
+func lookup(cmd *cobra.Command, args []string) {
+	vanityUrl, _ := cmd.Flags().GetString("vanityUrl")
+
+	log.WithFields(log.Fields{"command": cmd.Name(), "vanityUrl": vanityUrl}).Info("Provisioning group command")
+
+	response := callBackend(vanityUrl)
+
+	output.PrintCmdOutput(cmd, response)
+}
+
+func callBackend(vanityUrl string) any {
+	var response any
+	err := api.JSONGet(fmt.Sprintf("/provisioning/v1beta/tenants/lookup/vanityUrl/%s", vanityUrl), &response, nil)
```

Comment:
Another issue to consider: JSONGet requires a valid auth profile and will likely provide the auth headers. Does this API call require authentication / specific role? If not, we can indicate that this command may work with a auth=none profile (if you don't have access) but will also work with any configured auth type/tenant.

Reply:
Does this API call require authentication / specific role? - No, it doesn't.
If I do _./fsoc config set auth=none_ then I don't have URL in the context to do Platform API call.

Reply:
You can set `fsoc config create <PROFILE-NAME> auth=none url=<URL>` and it will still use that URL but not try to authenticate.

Reply:
I find it useful and added info to the long description of lookup command:
_Tenant lookup doesn't require valid authentication (auth=none) but any configured auth type/tenant will also work._
I also tested it but I didn't add it to documentation because don't want to make it more complicated to keep doc aligned with API changes. This works well:
./fsoc config set --profile my-profile auth=none url=https://tpinos-bot-14.saas.appd-test.com
./fsoc config use my-profile
./fsoc config get